### ISSUE TITLE
Applicability of SPOCK

### ISSUE THREAD

**Issue by zhexingli** [dtamayo/spock#14]
Hi Dan, great paper on SPOCK. I'm curious about the broader application of SPOCK. I know in the paper you mentioned that SPOCK can handle higher multiplicity planets as well as those not near MMR. But it seems all the cases discussed are about compact systems. Will SPOCK still work on non compact systems in general? If yes, does it have an edge over N-body or MEGNO in terms of speed and accuracy? Thanks.

**Comment by dtamayo** [comment]
Thanks! Yes, SPOCK is specific to compact systems. Part of the reason for that is that it effectively restricts to a particular path to instability (overlap of mean motion resonances). For more widely separated systems, or for longer timescales, overlap of secular resonances becomes the dominant source of chaos, which our model doesn't have features or training examples to capture. It generally gives right predictions for those sorts of systems--e.g., if you test it on the solar system it gives a high probability of stability, which is correct (there is a negligible chance of instability within 10^9 orbits). But if you push things by making e.g. eccentric hierarchical systems that do go unstable in Nbody within 10^9 orbits, I would not expect SPOCK to necessarily do well on them (would be an interesting test!). 

MEGNO is good in that it works in all cases, and it's great for mapping dynamical structure in phase space, but it's not clear how to calibrate it for testing for stability. It's telling you whether the system is chaotic on the timescale for which you ran your Nbody integration. If you want to test stability for 10^9 orbits, is it enough to run MEGNO over 10^3, 10^5, 10^7? If it tells you the system is chaotic, how likely is it to go unstable? If it tells you it's regular, how confident can you be that it won't be chaotic on longer timescales and go unstable? In all cases, I would definitely recommend running some comparisons against Nbody to make sure your conclusions are robust.

**Comment by zhexingli** [comment]
Thanks Dan! Is 10^9 orbits (I assume for the outer most planet?) the typical number to run to test stability?

**Comment by dtamayo** [comment]
We always define it for the innermost planet (though the distinction clearly isn't a huge deal for compact systems). As for the right number to pick, it depends on your application. 10^9 was a useful compromise in that it's starting to get comparable to the ages of observed systems, while still being short enough to be feasible to generate our training set. We discuss the implications of this choice, and how you might choose your own in more detail in 2.3 of our follow up paper: https://arxiv.org/abs/2009.11831. 

SPOCK is specifically trained for a fixed 10^9 orbits. We are also putting up on the arXiv tomorrow night a new deep learning model that predicts an instability time (rather than stable or not over 10^9 orbits). Look out for that one!

**Comment by zhexingli** [comment]
Great! Thanks Dan, looking forward to the new paper!
